test/e2e/operator_test.go

```diff
@@ -421,6 +420,31 @@ func componentRefEventuallyExists(w watch.Interface, exists bool, ref *corev1.Ob
 	}))
 }
 
+func multipleComponentRefsEventuallyExists(w watch.Interface, exists bool, refs []*corev1.ObjectReference) {
+	deadline, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer cancel()
+	founds := make([]bool, len(refs))
+
+	awaitPredicates(deadline, w, operatorPredicate(func(op *operatorsv1.Operator) bool {
+		if op.Status.Components == nil {
+			return false
+		}
+		for _, r := range op.Status.Components.Refs {
+			for i := 0; i < len(refs); i++ {
+				if !founds[i] && r.APIVersion == refs[i].APIVersion && r.Kind == refs[i].Kind && r.Namespace == refs[i].Namespace && r.Name == refs[i].Name {
+					founds[i] = true
+				}
+			}
+		}
+		for _, found := range founds {
+			if !found {
+				return !exists
+			}
+		}
```

Comment:
Nit: I found this a bit difficult to read.

At the cost of memory, we might be able to make this code more intuitive by creating a `map[corev1.ObjectReference]struct{}` that tracks each unseen ref and removing each key as it is seen. Once the length of the map is zero, we know that we've seen all elements.

Reply:
Thanks for review.  I updated the code.  I hope this is more readable than the previous one.